### DISCUSSION TITLE
graphics and tblr

### DISCUSSION THREAD

**Discussion by syvshc** [lvjr/tabularray#47] [Q&A]
I want to include some graphics in tblr. I'd like to put a figure and the text in 2 rows and let them align at center vertically. My English is poor, so here is the mwe:
```latex
\documentclass{article}
\usepackage{tabularray}
\usepackage{graphicx}
\begin{document}
    \begin{tblr}{
        vline{2},
        hlines,
        row{1}      =   {c, m},
        cell{1}{2}  =   {h},
        cell{2}{2}  =   {m}
    }
        text & \includegraphics[width = 2cm]{example-image-a}\\ 
        text & \includegraphics[width = 2cm]{example-image-b}\\     
    \end{tblr}
\end{document}
```
![image](https://user-images.githubusercontent.com/38098591/127767395-de0e57fc-0145-407c-ae68-1c6508360568.png)

I want something like the image-a, but image-a's top and bottom's sep seems not same. How can I do this?

**Comment by lvjr** [comment]
The extra space is from the `stretch`. You may set `stretch=0`:
````latex
\begin{tblr}{
    vline{2},
    hlines,
    row{1}      =   {c, m},
    cell{1}{2}  =   {h},
    cell{2}{2}  =   {m},
    stretch = 0,
}
    text & \includegraphics[width = 2cm]{example-image-a}\\ 
    text & \includegraphics[width = 2cm]{example-image-b}\\     
\end{tblr}
````
![image](https://user-images.githubusercontent.com/4239949/127767622-1f01fd02-9410-4425-848c-6cc425b8c403.png)

**Reply by syvshc** [reply]
Well, I wonder that why here `m` option doesn't work while `h` option works? or how to determine a figure's baseline?

**Reply by lvjr** [reply]
The baseline of a figure is always on the bottom edge.  And `tabularray` will always aligns two single-line `m`-cells in a row at the baseline (the same is true with traditional `tabular` environment). But `tabularray` doesn't align two `h` and `m` cells in a row.

**Reply by syvshc** [reply]
I see, thank you

**Comment by lvjr** [comment]
Another solution: use `\raisebox{-0.5\height}{\includegraphics{myfigure}}` to shift the baseline of the figure to the middle.

**Reply by muzimuzhi** [reply]
Alternatively, with [`graphbox`](https://ctan.org/pkg/graphbox) one can use `\includegraphics[align=c]{...}`.
